crossbar/worker/container.py

```diff
@@ -235,9 +238,15 @@ def start_container_component(self, id, config, reload_modules=False, details=No
             self._module_tracker.reload()
 
         # WAMP application session factory
-        #
+        # ultimately, this gets called once the connection is
+        # establised, from onOpen in autobahn/wamp/websocket.py:59
         def create_session():
-            return create_component(componentcfg)
+            try:
+                return create_component(componentcfg)
+            except Exception as e:
+                # AutobahnPython swallows exceptions from onOpen
+                log.err(_why="Instantiating component failed")
+                raise
 
```

Comment:
We could, I think, just call create_component() right here, and pass it to the factory via ``lambda: component`` instead of giving it ``create_session``; then we would get any instantiation errors immediately (and could, at least for this particular use-case, get rid of the 2-second stuff in controller/node.py)